patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -6125,4 +6176,46 @@
  			int type = list[Main.rand.Next(list.Count)];
  			Item item = new Item();
  			item.SetDefaults(type, false);
+@@ -40007,6 +_,41 @@
+ 					NetMessage.SendData(21, -1, -1, null, number, 1f, 0f, 0f, 0, 0, 0);
+ 				}
+ 			}
++		}
++
++		public int GetManaCost(Item item) {
++			int mana = (item.type != 127 || !this.spaceGun) ? (int)(item.mana * this.manaCost) : 0;
++			ItemLoader.GetManaCost(item, this, ref mana);
++			PlayerHooks.GetManaCost(this, item, ref mana);
++			return mana >= 0 ? mana : 0;
++		}
++
++		public bool CheckMana(Item item, int amount, bool pay = false, bool blockQuickMana = false) {
++			if (this.statMana >= amount) {
++				if (pay) {
++					ItemLoader.OnConsumeMana(item, this, amount);
++					PlayerHooks.OnConsumeMana(this, item, amount);
++					this.statMana -= amount;
++				}
++				return true;
++			}
++			if (blockQuickMana) {
++				return false;
++			}
++			ItemLoader.OnMissingMana(item, this, amount);
++			PlayerHooks.OnMissingMana(this, item, amount);
++			if (this.statMana < amount && this.manaFlower) {
++				this.QuickMana();
++			}
++			if (this.statMana >= amount) {
++				if (pay) {
++					ItemLoader.OnConsumeMana(item, this, amount);
++					PlayerHooks.OnConsumeMana(this, item, amount);
```

Comment:
I very recently added the `CombinedHooks` class, where I'd like to put any hooks of this style that bridge two modX types, could you just put a fowarder function there?

Reply:
Done.